src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.cs

```diff
@@ -33,6 +33,7 @@ internal enum TimeZoneInfoOptions
     [TypeForwardedFrom("System.Core, Version=3.5.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
     public sealed partial class TimeZoneInfo : IEquatable<TimeZoneInfo?>, ISerializable, IDeserializationCallback
     {
+        internal static bool Unsorted { get; } = AppContextConfigHelper.GetBooleanConfig("System.TimeZoneInfo.SystemTimeZonesUnsorted", "DOTNET_SYSTEM_TIME_ZONE_INFO_SYSTEM_TIME_ZONES_UNSORTED");
```

Comment:
I do not think we should be introducing this config switch. The config switch has major usability issues.

If we believe that it is important to avoid the sorting for perf, we should introduce the API overload that skips the sorting as proposed in the issue.

Reply:
Because the default behavior of the switch when unset is to go with the sorted variation, I was imagining that only the users with performance in mind would look for the switch and enable it. As for when it would be safe to set, if this switch just changes the order of the timezones in the collection, wouldn't it be on the user's responsibility if they found the switch and enabled it?

Reply:
I don't mind switching to the API overload. I was curious from @tarekgh's [comment](https://github.com/dotnet/runtime/issues/88691#issuecomment-1631350945), would it be a longer process because it would go into API review as it changes a documented API?

Reply:
Ah I see, it would affect all instances of `GetSystemTimeZones` because it would either be an app config or environment variable set. Whereas if we switched to an overload, it would only affect the instances where the overload is supplied as `false`?

Reply:
Right